libs/utils/include/utils/Panic.h

```diff
@@ -544,6 +545,27 @@ class TPanicStream : public PanicStream {
     }
 };
 
+template<typename PanicType>
+class FlagGuardedStream : public PanicStream {
+public:
+    FlagGuardedStream(bool enable, char const* function, char const* file, int line,
+            char const* condition)
+        : PanicStream(function, file, line, condition),
+          mEnablePanic(enable) {}
+    ~FlagGuardedStream() noexcept(false) {
+        if (mEnablePanic) {
+            PanicType::panic(mFunction, mFile, mLine, mLiteral, mStream.c_str());
+        } else {
+            LOG(WARNING) << "Failed [" << mLiteral << "]" << "\n  file=" << mFile
+                         << "\n  function=" << mFunction << "\n  line=" << mLine << "\n  info=\""
+                         << mStream.c_str() << "\"";
```

Comment:
This is not dependent on `PanicType` so there is no need to inline it. Basically I think the `else` should be a function call implemented in the CPP.

Reply:
done